test/reduce.jl

```diff
@@ -195,6 +195,11 @@ prod2(itr) = invoke(prod, (Any,), itr)
 @test all(x->x>0, [4]) == true
 @test all(x->x>0, [-3, 4, 5]) == false
 
+@test reduce(|, fill(trues(5), 3))  == trues(5)
+@test reduce(|, fill(falses(5), 3)) == falses(5)
+@test reduce(&, fill(trues(5), 3))  == trues(5)
+@test reduce(&, fill(falses(5), 3)) == falses(5)
```

Comment:
I don't think the issue was triggered until the array was at least 16 long, so this test probably passed before as well.